src/App.tsx

```diff
@@ -1,24 +1,136 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import classNames from 'classnames';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from './types/Todo';
+import { Filters } from './utils/Filters';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import {
+  deleteTodo,
+  getTodos,
+} from './api/todos';
+import { countTodos } from './utils/countTodos';
+import {
+  useTodos,
+} from './components/Contexts/TodosContext';
+import { USER_ID } from './utils/userToken';
+import {
+  useLoadingTodos,
+} from './components/Contexts/LoadingTodosContext';
+import {
+  useErrorMessage,
+} from './components/Contexts/ErrorMessageContext';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const { todos, setTodos } = useTodos();
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const { setLoadingTodos } = useLoadingTodos();
+  const {
+    errorMessage,
+    isErrorHidden,
+    setIsErrorHidden,
+    handleShowError,
+  } = useErrorMessage();
+
+  const [filterParam, setFilterParam] = useState(Filters.All);
+
+  const clearCompletedTodos = async () => {
+    const completedTodos = countTodos(todos, true).map(({ id }) => (
```

Comment:
Better to keep all `state`s together, all your own hooks also together, all constants together and all `useEffect`s also together